etna/models/nn/deepar_new/deepar.py

```diff
@@ -0,0 +1,367 @@
+from typing import Any
+from typing import Dict
+from typing import Iterator
+from typing import Optional
+from typing import Union
+
+import numpy as np
+import pandas as pd
+from typing_extensions import TypedDict
+
+from etna import SETTINGS
+from etna.distributions import BaseDistribution
+from etna.distributions import FloatDistribution
+from etna.distributions import IntDistribution
+
+if SETTINGS.torch_required:
+    import torch
+    import torch.nn as nn
+
+    from etna.models.base import DeepBaseModel
+    from etna.models.base import DeepBaseNet
+    from etna.models.nn.deepar_new.loss import GaussianLoss
+    from etna.models.nn.deepar_new.loss import NegativeBinomialLoss
+
+
+class DeepARBatchNew(TypedDict):
+    """Batch specification for DeepAR."""
+
+    encoder_real: "torch.Tensor"
+    decoder_real: "torch.Tensor"
+    encoder_target: "torch.Tensor"
+    decoder_target: "torch.Tensor"
+    segment: "torch.Tensor"
+    weight: "torch.Tensor"
+
+
+class DeepARNetNew(DeepBaseNet):
+    """DeepAR based Lightning module with LSTM cell."""
+
+    def __init__(
+        self,
+        input_size: int,
+        num_layers: int,
+        hidden_size: int,
+        lr: float,
+        scale: bool,
+        n_samples: int,
+        loss: Union[GaussianLoss, NegativeBinomialLoss],
+        optimizer_params: Optional[dict],
+    ) -> None:
+        """Init DeepAR.
+
+        Parameters
+        ----------
+        input_size:
+            size of the input feature space: target plus extra features
+        num_layers:
+            number of layers
+        hidden_size:
+            size of the hidden state
+        lr:
+            learning rate
+        scale:
+            if True, scale target values in batch before training by :math:`1 + mean`, where :math:`mean` is mean of target values in batch
+        n_samples:
+            if 1, return theoretical mean of distribution as a predicted value. if greater than 1, return empirical mean of `n_samples`
+        loss:
+            loss function
+        optimizer_params:
+            parameters for optimizer for Adam optimizer (api reference :py:class:`torch.optim.Adam`)
+        """
+        super().__init__()
+        self.save_hyperparameters()
+        self.num_layers = num_layers
+        self.input_size = input_size
+        self.hidden_size = hidden_size
+        self.rnn = nn.LSTM(
+            num_layers=self.num_layers, hidden_size=self.hidden_size, input_size=self.input_size, batch_first=True
+        )
+        self.linear_1 = nn.Linear(in_features=self.hidden_size, out_features=1)
```

Comment:
Better to create [ModuleDict](https://pytorch.org/docs/stable/generated/torch.nn.ModuleDict.html) with named projectors, now the naming is not the best one